test/e2e/serial/workload_placement_test.go

```diff
@@ -1037,6 +1037,82 @@ var _ = Describe("[serial][disruptive][scheduler] workload placement", func() {
 				return ok
 			}, time.Minute, time.Second*5).Should(BeTrue(), "resources not restored on %q", updatedPod.Spec.NodeName)
 		})
+
+		It("[test_id:47584] should be able to schedule guarnteed pod in selective way", func() {
+			nrtList := nrtv1alpha1.NodeResourceTopologyList{}
+			nrtListInitial, err := e2enrt.GetUpdated(fxt.Client, nrtList, time.Minute)
+			Expect(err).ToNot(HaveOccurred())
+
+			workers, err := e2enodes.GetWorkerNodes(fxt.Client)
+			Expect(err).ToNot(HaveOccurred())
+
+			// TODO choose randomly
+			targetedNodeName := workers[0].Name
+
+			nrtInitial, err := e2enrt.FindFromList(nrtListInitial.Items, targetedNodeName)
+			Expect(err).ToNot(HaveOccurred())
+
+			testPod := objects.NewTestPodPause(fxt.Namespace.Name, "testpod")
+			pSpec := &testPod.Spec
+
+			By(fmt.Sprintf("explicitly mentioning which we want pod to land on node %q", targetedNodeName))
+			pSpec.NodeName = targetedNodeName
+
+			By("setting a fake schedule name under the pod to make sure pod not scheduled by any scheduler")
+			noneExistingSchedulerName := "foo"
+			testPod.Spec.SchedulerName = noneExistingSchedulerName
+
+			cnt := &testPod.Spec.Containers[0]
+			requiredRes := corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("2"),
+				corev1.ResourceMemory: resource.MustParse("100Mi"),
+			}
+			cnt.Resources.Requests = requiredRes
+			cnt.Resources.Limits = requiredRes
+
+			By(fmt.Sprintf("creating pod %s/%s", testPod.Namespace, testPod.Name))
+			err = fxt.Client.Create(context.TODO(), testPod)
+			Expect(err).ToNot(HaveOccurred())
+
+			updatedPod, err := e2ewait.ForPodPhase(fxt.Client, testPod.Namespace, testPod.Name, corev1.PodRunning, timeout)
+			Expect(err).ToNot(HaveOccurred())
+
+			By(fmt.Sprintf("checking the pod landed on the target node %q vs %q", updatedPod.Spec.NodeName, targetedNodeName))
```

Comment:
maybe also check the pod was NOT scheduled by any scheduler after it got running?

Reply:
I thought about that, but it's a bit complex to check.
The thing is that this information mentioned implicitly under the events.

For example, if the pod was scheduled by a scheduler we'll see the following under the pod's events:
```
Events:
  Type    Reason          Age   From                  Message
  ----    ------          ----  ----                  -------
  Normal  Scheduled       24s   topo-aware-scheduler  Successfully assigned default/pod1 to <node-name>
    Normal  AddedInterface  23s   multus                Add eth0 [ip] from ovn-kubernetes
  Normal  Pulling         23s   kubelet               Pulling image "quay.io..."
  Normal  Pulled          22s   kubelet               Successfully pulled image "quay.io..." in 1.025265706s
  Normal  Created         22s   kubelet               Created container c1
  Normal  Started         22s   kubelet               Started container c1

```
But, if the pod is scheduled manually you'll see no line with reason named ```Scheduled```
But I'm not sure search for no line named ```Scheduled``` is robust enough.

Instead, I assigned a fake scheduler name. if the pod was relying on the scheduler, it would stay in ```Pending``` state since such scheduler doesn't exist.